### PR TITLE
[RSDK-7087] Display failed fragment mod updates in robot logs

### PR DESCRIPTION
**Pull request by piokasar**
https://viam.atlassian.net/browse/RSDK-7087

### CODE REVIEW

**Review by Kschappacher — APPROVED** [review]
LGTM but i would tag someone from @viamrobotics/netcode to review as well

File: config/proto_conversions.go [hunk 1]
```diff
@@ -85,6 +85,8 @@ func FromProto(proto *pb.RobotConfig, logger logging.Logger) (*Config, error) {
 		cfg.Debug = *proto.Debug
 	}
 
+	logAnyFragmentOverwriteErrors(logger, proto.OverwriteFragmentStatus)
```
Comment (benjirewis) [inline config/proto_conversions.go:88]: How does `proto.OverwriteFragmentStatus` get filled in?

File: config/proto_conversions.go [hunk 2]
```diff
@@ -905,3 +907,11 @@ func PackageTypeToProto(t PackageType) (*packagespb.PackageType, error) {
 		return packagespb.PackageType_PACKAGE_TYPE_UNSPECIFIED.Enum(), errors.Errorf("unknown package type %q", t)
 	}
 }
+
+func logAnyFragmentOverwriteErrors(logger logging.Logger, overwriteFragmentStatus []*pb.AppValidationStatus) {
+	if overwriteFragmentStatus != nil {
+		for _, status := range overwriteFragmentStatus {
+			logger.Errorw("error in overwriting fragment: ", "error", status.GetError())
```
Comment (cheukt) [inline config/proto_conversions.go:914]:
```suggestion
			logger.Errorw("error in overwriting fragment", "error", status.GetError())
```
nit: no need for semicolon

**Review by cheukt — COMMENTED** [review]
curious - what do the logs look like with this change?

**Review by michaellee1019 — APPROVED** [review]

File: config/proto_conversions.go (hunk 1 above)
Comment (piokasar) [reply config/proto_conversions.go:88]: Thanks for looking at this so quickly!
It gets filled in [here](https://github.com/viamrobotics/app/blob/93c939fc8f142620812e4f54e61effb56c90f58b/robotservice/config_retriever.go#L299).

Specifically, when there is an error in overwriting a fragment whether it's a user error or an internal error.

Comment (benjirewis) [reply config/proto_conversions.go:88]: Cool; thank you! Agree with @cheukt then about the colon but otherwise LGTM :)

**Comment by piokasar** [comment]
> curious - what do the logs look like with this change?

here's an example (with the semicolon removed) when there is an error in the fragment_mods a user provides:

```
error in overwriting fragment	{"error":"mods command(s) {\"$unset\":{\"services.dmsv2.attributes.sync_interval_mins\":0.6}} has an error and did not modify the fragment with ID e057ebc6-c6dc-450b-9473-0703c5ddfbb1 with url \"https://app.viam.com/fragment?id=e057ebc6-c6dc-450b-9473-0703c5ddfbb1\""}
```

I think we'd probably have to strip the backslashes on the app side.

**Comment by piokasar** [comment]
> looks fine to me, but `mods command(s)` don't mean anything to me, so could be clarified a bit more in app? agree with stripping backslashes in app

ok yup! happy to do. that will require a change in the gcloud function, and the stripping the backslashes will require a change in app. I'll tag you on that .

**Pull request merged**